libs/server/Storage/Functions/MainStore/VarLenInputMethods.cs

```diff
@@ -185,6 +185,13 @@ public int GetRMWModifiedValueLength(ref SpanByte t, ref SpanByte input)
                         // No additional allocation needed.
                         break;
 
+                    case RespCommand.GETEX:
+                        if (input.ExtraMetadata > 0)
```

Comment:
Assuming this should always be true if we reach here? Could just do a Debug.Assert(input.ExtraMetadata > 0)

Reply:
Done

Reply:
I am marking it as unresolved as I have an open question. Do we need this `if` check since we have CopyUpdater now?

Reply:
Hmm interesting, the current `PERSIST` doesn't do that. Anyway, I don't want to get confused even more. I will do it as you said. just return `sizeof(int) + t.Length` when `input.ExtraMetadata == 0`

Reply:
If we (1) remove the assert; and (2) simply return `sizeof(int) + t.LengthWithoutMetadata + input.MetadataSize` -- that should cover all cases, right?

Reply:
@badrishc  Wow, yes. Never even thought of it. Will push the change soon, in middle of fixing a bug with GeoSearch.

Reply:
@badrishc Can you explain to me why the current `PERSIST` doesn't return any length? Shouldn't it return `sizeof(int) + t.LengthWithoutMetadata`?

Reply:
You are right - this looks like a bug in persist that should be fixed. Good catch.

Reply:
Fixed